hepcrawl/spiders/elsevier.py

```diff
@@ -0,0 +1,235 @@
+                self.packages_bucket_name
+            )
+            self.s3_files_bucket_conn = self.s3_bucket_connection(
+                self.files_bucket_name
+            )
+            self.s3_client = self.connect_s3_client()
+
+    def create_s3_connection(self):
+        session = boto3.Session(
+            aws_access_key_id=self.access_key_id,
+            aws_secret_access_key=self.secret_access_key,
+        )
+        s3 = session.resource("s3", endpoint_url=self.s3_host)
+        return s3
+
+    def s3_bucket_connection(self, bucket_name):
+        bucket_connection = self.s3_connection.Bucket(bucket_name)
+        return bucket_connection
+
+    def connect_s3_client(self):
+        s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=self.access_key_id,
+            aws_secret_access_key=self.secret_access_key,
+            endpoint_url=self.s3_host,
+        )
+        return s3_client
+
+    def create_presigned_url(self, bucket, file, method):
+        url = self.s3_client.generate_presigned_url(
+            ClientMethod=method,
+            Params={"Bucket": bucket, "Key": file},
+            ExpiresIn=920000,
+        )
+        return url
+
+    def _get_keys_names_from_bucket(self):
+        keys = set([key.key for key in self.s3_packages_bucket_conn.objects.all()])
+        return keys
+
+    def _get_package_urls_from_elsevier(self, elsevier_metadata):
+        """
+        Extracts names and urls of the zip packages from elsevier batch feed
+
+        Returns:
+            dict(name: url): dict of zip packages names and urls
+
+        """
+        packages_metadata_parsed = et.fromstring(elsevier_metadata)
+        urls_for_packages = {}
+        for children in packages_metadata_parsed.getchildren():
+            if "entry" in children.tag:
+                file_data = children.getchildren()
+                link = file_data[1].attrib["href"]
+                urls_for_packages[file_data[0].text] = link
+        return urls_for_packages
+
+    def _get_all_new_packages(self, elsevier_metadata):
+        """
+        Checks which packages from elsevier batch feed are not in the s3 bucket yet
+        Returns:
+            dict(name: url): dict of zip packages names and urls
+
+        """
+        urls_for_packages = self._get_package_urls_from_elsevier(elsevier_metadata)
+        bucket_data = self._get_keys_names_from_bucket()
+        packages_not_in_bucket = {
+            name: urls_for_packages[name]
+            for name in set(urls_for_packages.keys()) - bucket_data
+        }
+        self.new_packages = set(packages_not_in_bucket.keys())
+        return packages_not_in_bucket
+
+    def start_requests(self):
+        elsevier_batch_download_url = self.elsevier_consyn_url
+        yield Request(
+            elsevier_batch_download_url, callback=self.get_packages_from_elsevier
+        )
+
+    def get_packages_from_elsevier(self, response):
```

Comment:
```suggestion
    def get_packages_from_consyn_feed(self, response):
```
the class and the file are called `elsevier`, no need to repeat it all the time, better use more descriptive names.